tools/hansel/bio_hansel.xml

```diff
@@ -1,227 +1,269 @@
+#set $_reverse_ext = '.fastq.gz' if $re.match(r'.*\.gz$', $input.reverse.name) else '.fastq'
+#set $_reverse = '"{}_2{}"'.format($re.sub(r'_[12]\..+$', '', $input.reverse.name), $_reverse_ext)
+#set $input_files = '{} {}'.format($_forward, $_reverse)
+  ln -s "$input.forward" $_forward &&
+  ln -s "$input.reverse" $_reverse &&
+#elif $input.type == 'single'
+#set $input_files = '"{}"'.format($input.single.name)
+  ln -s "$input.single" $input_files &&
+#elif $input.type == 'paired_collection'
+#set $_forward_ext = '.fastq.gz' if $re.match(r'.*\.gz$', str($input.paired_collection.forward)) else '.fastq'
+#set $_forward = '"{}_1{}"'.format($input.paired_collection.name, $_forward_ext)
+#set $_reverse_ext = '.fastq.gz' if $re.match(r'.*\.gz$', str($input.paired_collection.reverse)) else '.fastq'
+#set $_reverse = '"{}_2{}"'.format($input.paired_collection.name, $_reverse_ext)
+#set $input_files = '{} {}'.format($_forward, $_reverse)
+  ln -s "$input.paired_collection.forward" $_forward &&
+  ln -s "$input.paired_collection.reverse" $_reverse &&
+#end if
+
+## Checking for custom scheme.
+#if $type_of_scheme.scheme_type == "custom":
+  #if $type_of_scheme.scheme_input.is_of_type('fasta'):
+    ln -s '$type_of_scheme.scheme_input' '$type_of_scheme.scheme_input.name' &&
+  #end if
+#end if
+
+hansel
+  -vvv
+  -t "\${GALAXY_SLOTS:-1}"
+  -o results.tab
+  -O match_results.tab
+  -S tech_results.tab
+  $dev_args.use_json
+  $input_files
+  --scheme
+#if $type_of_scheme.scheme_type == "custom":
+  '$type_of_scheme.scheme_input.name'
+#else:
+  $type_of_scheme.scheme_type
+#end if
+#if $kmer_vals.kmer_min
+  --min-kmer-freq $kmer_vals.kmer_min
+#end if
+#if $kmer_vals.kmer_max
+  --max-kmer-freq $kmer_vals.kmer_max
+#end if
+#if $qc_vals.low_cov_depth_freq
+  --low-cov-depth-freq $qc_vals.low_cov_depth_freq
+#end if
+#if $qc_vals.max_missing_tiles
+  --max-missing-tiles $qc_vals.max_missing_tiles
+#end if
+#if $qc_vals.min_ambiguous_tiles
+  --min-ambiguous-tiles $qc_vals.min_ambiguous_tiles
+#end if
+#if $qc_vals.max_intermediate_tiles
+  --max-intermediate-tiles $qc_vals.max_intermediate_tiles
+#end if
+#if $qc_vals.low_coverage_warning
+  --low-cov-warning $qc_vals.low_coverage_warning
+#end if
+]]>
+  </command>
+  <inputs>
+    <conditional name="input">
+      <param name="type" type="select" label="Sequence input type">
+        <option value="fasta">FASTA</option>
+        <option value="paired">Paired-end FASTQs</option>
+        <option value="single">Single-end FASTQ</option>
+        <option value="paired_collection">Paired-end FASTQ collection</option>
+      </param>
+      <when value="fasta">
+        <param name="fasta"
+          type="data" format="fasta"
+          optional="false"
+          label="FASTA file"
+          />
+      </when>
+      <when value="paired">
+        <param name="forward"
+          type="data" format="fastq,fastqsanger,fastqillumina,fastqsolexa"
```

Comment:
@Takadonet so does this mean you don't need to add `fastqsanger.gz` support?

Reply:
I borrowed that list of `format`s from Mentalist. I don't think most of those are needed so I can take them out. The quality score encoding shouldn't make a difference for bio_hansel results since it doesn't use them. 

Yes, I've added the `gz` formats in 25438b0 since they are supported!

Reply:
You don't have to specify fastqsanger.gz support directly for you. It should not hurt to have explicitly marked down.

Reply:
Cool.  Leaving it in is okay with me.  Thanks @Takadonet and @peterk87